### PR TITLE
fixes gh-1304 add index table to popen

### PR DESCRIPTION
**Pull request by Onvember**

### CODE REVIEW

File: doc/reference/reference_lua/popen.rst [hunk 1]
```diff
@@ -32,6 +32,54 @@ object:
 Either function returns a handle which we will call ``popen_handle`` or ``ph``.
 With the handle one can execute methods.
 
+===============================================================================
+                                    Index
+===============================================================================
+
+Below is a list of all ``popen`` functions.
```
Comment (Totktonada) [inline doc/reference/reference_lua/popen.rst:39]: Technically module functions and handle methods are listed here, not only functions.

Should not also  **Handle fields** and **Module constants** sections be listed here?

File: doc/reference/reference_lua/popen.rst [hunk 2]
```diff
@@ -32,6 +32,54 @@ object:
...
+    | <popen-shell>`                       |                                 |
+    +--------------------------------------+---------------------------------+
+    | :ref:`popen.new()                    | Execute a child program in      |
+    | <popen-new>`                         | a new process                   |
+    +--------------------------------------+---------------------------------+
+    | :ref:`popen_handle:read()            | Read data from a child peer     |
+    | <popen-read>`                        |                                 |
+    +--------------------------------------+---------------------------------+
+    | :ref:`popen_handle:write()           | Write string str to stdin       |
+    | <popen-write>`                       | stream of a child process       |
```
Comment (Totktonada) [inline doc/reference/reference_lua/popen.rst:59]: Typo: `string str` → `a string`.

Comment (Onvember) [reply doc/reference/reference_lua/popen.rst]: fixed in 61ad4a5

File: doc/reference/reference_lua/popen.rst (hunk 1 above)
Comment (Onvember) [reply doc/reference/reference_lua/popen.rst]: Fixed in 61ad4a5

**Pull request merged**